underlay/src/main/java/bio/terra/tanagra/filterbuilder/impl/core/FilterableGroupFilterBuilder.java

```diff
@@ -0,0 +1,192 @@
+package bio.terra.tanagra.filterbuilder.impl.core;
+
+import static bio.terra.tanagra.utils.ProtobufUtils.deserializeFromJsonOrProtoBytes;
+
+import bio.terra.tanagra.api.filter.AttributeFilter;
+import bio.terra.tanagra.api.filter.BooleanAndOrFilter;
+import bio.terra.tanagra.api.filter.BooleanAndOrFilter.LogicalOperator;
+import bio.terra.tanagra.api.filter.EntityFilter;
+import bio.terra.tanagra.api.filter.TextSearchFilter;
+import bio.terra.tanagra.api.filter.TextSearchFilter.TextSearchOperator;
+import bio.terra.tanagra.api.shared.BinaryOperator;
+import bio.terra.tanagra.api.shared.Literal;
+import bio.terra.tanagra.api.shared.NaryOperator;
+import bio.terra.tanagra.exception.InvalidQueryException;
+import bio.terra.tanagra.filterbuilder.EntityOutput;
+import bio.terra.tanagra.filterbuilder.FilterBuilder;
+import bio.terra.tanagra.filterbuilder.impl.core.utils.AttributeSchemaUtils;
+import bio.terra.tanagra.filterbuilder.impl.core.utils.EntityGroupFilterUtils;
+import bio.terra.tanagra.proto.criteriaselector.configschema.CFFilterableGroup;
+import bio.terra.tanagra.proto.criteriaselector.dataschema.DTFilterableGroup;
+import bio.terra.tanagra.underlay.Underlay;
+import bio.terra.tanagra.underlay.entitymodel.Entity;
+import bio.terra.tanagra.underlay.entitymodel.entitygroup.EntityGroup;
+import bio.terra.tanagra.underlay.entitymodel.entitygroup.GroupItems;
+import bio.terra.tanagra.underlay.uiplugin.CriteriaSelector;
+import bio.terra.tanagra.underlay.uiplugin.SelectionData;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import org.apache.commons.lang3.NotImplementedException;
+import org.apache.commons.lang3.StringUtils;
+
+public class FilterableGroupFilterBuilder
+    extends FilterBuilder<CFFilterableGroup.FilterableGroup, DTFilterableGroup.FilterableGroup> {
+
+  public FilterableGroupFilterBuilder(CriteriaSelector criteriaSelector) {
+    super(criteriaSelector);
+  }
+
+  @Override
+  public EntityFilter buildForCohort(Underlay underlay, List<SelectionData> selectionData) {
+    if (selectionData.size() != 1) {
+      throw new InvalidQueryException(
+          "Filterable group filter builder does not support modifiers.");
+    }
+
+    DTFilterableGroup.FilterableGroup data = deserializeData(selectionData.get(0).getPluginData());
+    if (data == null) { // Empty selection data = null filter for a cohort.
+      return null;
```

Comment:
Returning `null` here makes an empty selection do nothing. What it should actually do is filter for participants who have any variant data. I think adding a `|| filtersToOr.isEmpty()` on line 89 might be enough.

Reply:
sg, done